core/sdk-core/src/main/java/software/amazon/awssdk/core/FileRequestBodyConfiguration.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    /**
+     * Create a {@link Builder}, used to create a {@link FileRequestBodyConfiguration}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    /**
+     * @return the size of each chunk to read from the file
+     */
+    public Integer chunkSizeInBytes() {
+        return chunkSizeInBytes;
+    }
+
+    /**
+     * @return the file position at which the request body begins.
+     */
+    public Long position() {
+        return position;
+    }
+
+    /**
+     * @return the number of bytes to read from this file.
+     */
+    public Long numBytesToRead() {
+        return numBytesToRead;
+    }
+
+    /**
+     * @return the file path
+     */
+    public Path path() {
+        return path;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        FileRequestBodyConfiguration that = (FileRequestBodyConfiguration) o;
+
+        if (!Objects.equals(chunkSizeInBytes, that.chunkSizeInBytes)) {
+            return false;
+        }
+        if (!Objects.equals(position, that.position)) {
+            return false;
+        }
+        if (!Objects.equals(numBytesToRead, that.numBytesToRead)) {
+            return false;
+        }
+        return Objects.equals(path, that.path);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = chunkSizeInBytes != null ? chunkSizeInBytes.hashCode() : 0;
+        result = 31 * result + (position != null ? position.hashCode() : 0);
+        result = 31 * result + (numBytesToRead != null ? numBytesToRead.hashCode() : 0);
+        result = 31 * result + (path != null ? path.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder(this);
+    }
+
+    public interface Builder extends CopyableBuilder<Builder, FileRequestBodyConfiguration> {
+
+        /**
+         * Sets the {@link Path} to the file containing data to send to the service
+         *
+         * @param path Path to file to read.
+         * @return This builder for method chaining.
+         */
+        Builder path(Path path);
+
+        /**
+         * Sets the size of chunks read from the file. Increasing this will cause more data to be buffered into memory but
+         * may yield better latencies. Decreasing this will reduce memory usage but may cause reduced latency. Setting this value
+         * is very dependent on upload speed and requires some performance testing to tune.
+         *
+         * <p>The default chunk size is 16 KiB</p>
+         *
+         * @param chunkSize New chunk size in bytes.
+         * @return This builder for method chaining.
+         */
+        Builder chunkSizeInBytes(Integer chunkSize);
```

Comment:
I kept Integer here because we only support reading up to `Integer.MAX_VALUE` number of bytes per chunk for now (ByteBuffer only takes int) and I don't think we should support long for this use case. LMK if you think otherwise.

Reply:
2 GB per chunk is probably a reasonable maximum for now, if it's not easy to support a higher number.